### PR TITLE
Add dvorak's spanish variant to quantum/keymap_extras

### PR DESCRIPTION
**Pull request by joseandres42**
## Description

Based on keymap_spanish.h, I created a keymap for the spanish version of the Dvorak layout.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/* Copyright 2015-2016 Jack Humbert
```
Comment (fauxpark) [inline quantum/keymap_extras/keymap_spanish_dvorah.h:1]: Please change the copyright here to yourself :)

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+#define DV_EQL  S(DV_0)    // =
+#define DV_QUES S(DV_QUOT) // ?
+#define DV_IQUE S(DV_IEXL) // ¿
+// Row 2
+#define DV_COLN S(KC_DOT)  // :
+#define DV_SCLN S(KC_COMM) // ;
+#define DV_CIRC S(DV_GRV)  // ^
+#define DV_ASTR S(DV_PLUS) // *
+// Row 3
+#define DV_DIAE S(DV_GRV)  // ¨
```
Comment (vomindoraan) [inline quantum/keymap_extras/keymap_spanish_dvorah.h:120]: Is this a dead key (combining diaeresis accent)?
```suggestion
#define DV_DIAE S(DV_GRV)  // ¨ (dead)
```

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+ * │ º │ 1 │ 2 │ 3 │ 4 │ 5 │ 6 │ 7 │ 8 │ 9 │ 0 │ ' │ ¡ │       │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+ * │     │ . │ , │ Ñ │ P │ Y │ F │ G │ C │ H │ L │ ` │ + │     │
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐    │
+ * |      │ A │ O │ E │ U │ I │ D │ R │ T │ N │ S │ ' │ Ç │    │
+ * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴────┤
+ * │    │ < │ - │ Q │ J │ K │ X │ B │ M │ W │ V │ Z │          │
+ * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
```
[the remark this answers is not in the document]
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: No problem, done! And sorry for the typo :)

File: quantum/keymap_extras/keymap_spanish_dvorah.h (hunk 1 above)
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: Done! :)

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+#define DV_CIRC S(DV_GRV)  // ^
+#define DV_ASTR S(DV_PLUS) // *
+// Row 3
+#define DV_DIAE S(DV_GRV)  // ¨
+// Row 4
+#define DV_RABK S(DV_LABK) // >
+#define DV_UNDS S(DV_MINS) // _
+
+
+
```
[the remark this answers is not in the document]
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: It is indeed unnecessary, removed!

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+#define DV_COMM KC_W    // ,
+#define DV_NTIL KC_E    // Ñ
+#define DV_P    KC_R    // P
+#define DV_Y    KC_T    // Y
+#define DV_F    KC_Y    // F
+#define DV_G    KC_U    // G
+#define DV_C    KC_I    // C
+#define DV_H    KC_O    // H
+#define DV_L    KC_P    // L
+#define DV_GRV  KC_LBRC // `
```
[the remark this answers is not in the document]
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: It is, fixed!

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+#define DV_O    KC_S    // O
+#define DV_E    KC_D    // E
+#define DV_U    KC_F    // U
+#define DV_I    KC_G    // I
+#define DV_D    KC_H    // D
+#define DV_R    KC_J    // R
+#define DV_T    KC_K    // T
+#define DV_N    KC_L    // N
+#define DV_S    KC_SCLN // S
+#define DV_ACUT KC_QUOT // '
```
[the remark this answers is not in the document]
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: It is an acute accent and a dead key, fixed.

File: quantum/keymap_extras/keymap_spanish_dvorah.h (hunk 2 above)
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: It is a dead key, fixed.

File: quantum/keymap_extras/keymap_spanish_dvorah.h [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+#define DV_SLSH S(DV_7)    // /
+#define DV_LPRN S(DV_8)    // (
+#define DV_RPRN S(DV_9)    // )
+#define DV_EQL  S(DV_0)    // =
+#define DV_QUES S(DV_QUOT) // ?
+#define DV_IQUE S(DV_IEXL) // ¿
+// Row 2
+#define DV_COLN S(KC_DOT)  // :
+#define DV_SCLN S(KC_COMM) // ;
+#define DV_CIRC S(DV_GRV)  // ^
```
[the remark this answers is not in the document]
Comment (joseandres42) [reply quantum/keymap_extras/keymap_spanish_dvorah.h]: It is a dead key, fixed.

**Comment by joseandres42** [comment]
I added the sendstring file (sendstring_spanish_dvorak.h). While making it I think I discovered an error in sendstring_spanish.h, because the ~ symbol is written with AltGr+4, not with AltGr+Ñ, as I think the file intended. Please someone confirm this.

**Comment by vomindoraan** [comment]
I checked, you are correct. Feel free to update that.

**Comment by joseandres42** [comment]
Thank you. I think everything is ready for merge now :)

File: quantum/keymap_extras/sendstring_spanish_dvorak.h [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Sendstring lookup tables for Spanish's dvorak layout
```
Comment (fauxpark) [inline quantum/keymap_extras/sendstring_spanish_dvorak.h:17]:
```suggestion
// Sendstring lookup tables for Spanish Dvorak layout
```

File: quantum/keymap_extras/sendstring_spanish.h [hunk 9]
```diff
@@ -96,5 +96,5 @@ const uint8_t ascii_to_keycode_lut[128] PROGMEM = {
     // p     q        r        s        t        u        v        w
     ES_P,    ES_Q,    ES_R,    ES_S,    ES_T,    ES_U,    ES_V,    ES_W,
     // x     y        z        {        |        }        ~        DEL
-    ES_X,    ES_Y,    ES_Z,    ES_ACUT, ES_1,    ES_CCED, ES_NTIL, KC_DEL
+    ES_X,    ES_Y,    ES_Z,    ES_ACUT, ES_1,    ES_CCED, ES_4, KC_DEL
```
Comment (fauxpark) [inline quantum/keymap_extras/sendstring_spanish.h:99]:
```suggestion
    ES_X,    ES_Y,    ES_Z,    ES_ACUT, ES_1,    ES_CCED, ES_4,    KC_DEL
```

File: quantum/keymap_extras/sendstring_spanish_dvorak.h [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    // X     Y        Z        [        \        ]        ^        _
+    DV_X,    DV_Y,    DV_Z,    DV_GRV,  DV_MORD, DV_PLUS, DV_GRV,  DV_MINS,
+    // `     a        b        c        d        e        f        g
+    DV_GRV,  DV_A,    DV_B,    DV_C,    DV_D,    DV_E,    DV_F,    DV_G,
+    // h     i        j        k        l        m        n        o
+    DV_H,    DV_I,    DV_J,    DV_K,    DV_L,    DV_M,    DV_N,    DV_O,
+    // p     q        r        s        t        u        v        w
+    DV_P,    DV_Q,    DV_R,    DV_S,    DV_T,    DV_U,    DV_V,    DV_W,
+    // x     y        z        {        |        }        ~        DEL
+    DV_X,    DV_Y,    DV_Z,    DV_ACUT, DV_1,    DV_CCED, DV_4, KC_DEL
```
Comment (fauxpark) [inline quantum/keymap_extras/sendstring_spanish_dvorak.h:99]:
```suggestion
    DV_X,    DV_Y,    DV_Z,    DV_ACUT, DV_1,    DV_CCED, DV_4,    KC_DEL
```

**Comment by fauxpark** [comment]
Sorry, one last thing, LGTM otherwise.

**Comment by joseandres42** [comment]
Can someone retrigger the Travis CI build? It seems apt-get failed in its build because a connection issue .

**Pull request merged**